packages/beacon-node/src/metrics/metrics/beacon.ts

```diff
@@ -121,6 +121,38 @@ export function createBeaconMetrics(register: RegistryMetricCreator) {
       buckets: [0.1, 1, 2, 4, 10],
       labelNames: ["source"],
     }),
+    executionBlockProductionTimeSteps: register.histogram<"step">({
+      name: "beacon_block_production_execution_steps_seconds",
+      help: "Detailed steps runtime of execution block production",
+      buckets: [0.01, 0.1, 0.2, 0.5, 1],
+      /**
```

Comment:
I feel like commenting those values here is not the right approach. We should use an enum in such cases as we did for block production source.

Might be worth to look into how we could make label values type safe.

Reply:
We have to change the types from pom-client for that. That's a bigger task to do. Feel free to create a separate issue to track in future.